app/action/search-actions.coffee

```diff
@@ -96,24 +81,125 @@ getPeliasDataOrEmptyArray = (input, geolocation) ->
+  searches = []
+  fav = favourites.map (f) -> '"' + f + '"'
+  searches.push('favouriteRoutes:routes(ids:[' + fav.join(',') + ']) {patterns {code} agency {name} shortName type longName}')
+
+  if input != undefined &&  input != null and input.trim().length > 0
+    doRouteSearch = doStopSearch = false
+
+    #dumb quess on what the user is trying to do
+    isNumber = input.match(/^\d+$/) != null
+    if isNumber
+      lnLen = input.match(/^\d+$/).length
+      if lnLen <= 3
+        doRouteSearch = true
+
+      if lnLen == 4
+        doStopSearch = true
+    else
+      doRouteSearch = doStopSearch = true
+
+  if doRouteSearch then searches.push 'routes(name:"' + input + '") {patterns {code} agency {name} shortName type longName}'
+  if doStopSearch then searches.push 'stops(name:"' + input + '") {gtfsId lat lon name}'
+
+  if searches.length > 0
+    suggestions = []
+    refLatLng = geoUtils.getLatLng(reference.lat, reference.lon)
+    return queryGraphQL('{' + searches.join(' ') + '}').then (response) ->
+
+      favouriteRoutes = mapRoutes(response?.data?.favouriteRoutes).map (favourite) ->
+        favourite.properties.layer = 'favourite'
+        favourite.type= "Favourite"
+        favourite
+      [].concat sortBy(favouriteRoutes, (item) -> ['agency.name', 'properties.label'])
+      .concat sortBy(mapStops(response?.data?.stops), (item) ->
+        stopPos = geoUtils.getLatLng(item.geometry.coordinates[1], item.geometry.coordinates[0])
+        Math.round(stopPos.distanceTo(refLatLng) / 50000))
+      .concat sortBy(mapRoutes(response?.data?.routes), (item) -> ['agency.name', 'properties.label'])
+
+  else
+    Promise.resolve []
+
+executeSearch = (actionContext, params) ->
+  processResults(actionContext, [])
+  {input, type} = params
   geoLocation = actionContext.getStore('PositionStore').getLocationState()
-  favouriteLocations = actionContext.getStore("FavouriteLocationStore").getLocations()
-  oldSearches = actionContext.getStore("OldSearchesStore").getOldSearches()
-  getPeliasDataOrEmptyArray(input, geoLocation)
-  .then addCurrentPositionIfEmpty
-  .then (suggestions) -> addFavouriteLocations(favouriteLocations, suggestions, input)
-  .then (suggestions) -> addOldSearches(oldSearches, suggestions, input)
-  .then sort
-  .then uniq
-  .then (suggestions) ->
-    processResults actionContext, suggestions
-  .catch (e) ->
-    console.error("error occurred", e)
-
-geocode =
-  debounce(directGeocode, 300)
-
-#used by the modal
+  referenceLocation = if geoLocation.hasLocation then {lon: geoLocation.lon, lat: geoLocation.lat} else console.log("no location, what's the reference?")
+
+  #endpoint
+  if type == 'endpoint'
+    favouriteLocations = actionContext.getStore("FavouriteLocationStore").getLocations()
+    oldSearches = actionContext.getStore("OldSearchesStore").getOldSearches()
+    Promise.all([getGeocodingResult(input, geoLocation), getEndpointGTFSResult (input)])
```

Comment:
I would say that no GTFS results here. Now we get routes for origin location, which is weird.